src/snowflake/connector/bind_upload_agent.py

```diff
@@ -38,13 +38,24 @@ def __init__(
             rows: Rows of binding parameters in CSV format.
             stream_buffer_size: Size of each file, default to 10MB.
         """
+        self._use_scoped_temp_object = (
+            cursor.connection._session_parameters.get(
+                _PYTHON_SNOWPARK_USE_SCOPED_TEMP_OBJECTS_STRING, False
+            )
+            if cursor.connection._session_parameters
+            else False
+        )
```

Comment:
what about the following?
```suggestion
        )
        self. _STAGE_NAME = (
            "SNOWPARK_TEMP_STAGE_BIND"
            if self._use_scoped_temp_object
            else "SYSTEMBIND"
        )
```